hardware/PhilipsHue/PhilipsHue.cpp

```diff
@@ -462,7 +462,7 @@ void CPhilipsHue::InsertUpdateSwitch(const int NodeID, const _eHueLightType LTyp
 	{
 		//Send as Lighting 2
 		char szID[10];
-		sprintf(szID, "%X%02X%02X%02X", 0, 0, 0, NodeID);
+		sprintf(szID, "%07X", (unsigned int)NodeID);
```

Comment:
Thanks for the patch!, Should this perhaps be %08X ? (as we use this a lot)

Reply:
For some reason Lighting 2 uses a 7 character DeviceID instead of 8 characters (as other hue light types do).  Things didn't work correctly when I used %08X.  Further, `sprintf(szID, "%X%02X%02X%02X", 0, 0, 0, NodeID);` seems to be used quite often to generate a 7 character DeviceID for Lighting 2 elsewhere in the code (e.g. [FritzboxTCP.cpp](https://github.com/domoticz/domoticz/blob/development/hardware/FritzboxTCP.cpp),  [Sterbox.cpp](https://github.com/domoticz/domoticz/blob/development/hardware/Sterbox.cpp), [RelayNet.cpp](https://github.com/domoticz/domoticz/blob/development/hardware/RelayNet.cpp)), so %07X does seem to be right, at least for Lighting 2.

The reason I changed it from  `sprintf(szID, "%X%02X%02X%02X", 0, 0, 0, NodeID);` to `sprintf(szID, "%07X", (unsigned int)NodeID);` was because the ID of Phillips Hue groups require 2 bytes instead of one.